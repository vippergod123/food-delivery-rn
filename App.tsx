import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import React from 'react';
import Tabs from './navigation/tabs';
import {OrderDelivery, Restaurant} from './screens';
import Home from './screens/Home';

const Stack = createStackNavigator();
const App = () => {
  return (
    <NavigationContainer>
      <Stack.Navigator
        screenOptions={{headerShown: false}}
        initialRouteName={'Home'}>
        <Stack.Screen name="Home" component={Tabs} />
        <Stack.Screen name="Restaurant" component={Restaurant} />
        <Stack.Screen name="OrderDelivery" component={OrderDelivery} />
      </Stack.Navigator>
    </NavigationContainer>
  );
};

export default App;
