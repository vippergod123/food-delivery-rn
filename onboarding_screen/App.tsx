import React from 'react';
import {createStackNavigator} from '@react-navigation/stack';
import {NavigationContainer} from '@react-navigation/native';
import 'react-native-gesture-handler';
import {Onboarding} from './app/screens';
const Stack = createStackNavigator();

const App = () => {
  return (
    <NavigationContainer>
      <Stack.Navigator initialRouteName="Onboarding">
        <Stack.Screen
          name="Onboarding"
          component={Onboarding}
          options={{headerShown: false}}
        />
      </Stack.Navigator>
    </NavigationContainer>
  );
};

export default App;
