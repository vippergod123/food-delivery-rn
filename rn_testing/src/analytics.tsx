export function track(name) {
  doSomething(name);
}
