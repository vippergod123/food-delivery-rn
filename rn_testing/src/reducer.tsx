const initialState = {
  msg: '',
};

function reducer(state = initialState, action) {
  switch (action.type) {
    case 'setMsg':
      return {...state, msg: action.payload};
    default:
      return state;
  }
}

export default reducer;
