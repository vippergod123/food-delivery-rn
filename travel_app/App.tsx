import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import React from 'react';
import Tabs from './navigation/tabs';
import {Dashboard, Place} from './screens';

const Stack = createStackNavigator();
const App = () => {
  return (
    <NavigationContainer>
      <Stack.Navigator
        initialRouteName="Dashboard"
        screenOptions={{headerShown: false}}>
        <Stack.Screen
          name="Dashboard"
          component={Tabs}
          options={{headerShown: false}}
        />
        <Stack.Screen name="Place" component={Place} />
      </Stack.Navigator>
    </NavigationContainer>
  );
};

export default App;
