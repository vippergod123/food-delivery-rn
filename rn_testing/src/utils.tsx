import {track} from './analytics';

export function addNumber(x, y) {
  return x + y;
}

export function testMockFuction(name) {
  return track(name);
}

export function sayBye() {
  return 'Bye';
}
