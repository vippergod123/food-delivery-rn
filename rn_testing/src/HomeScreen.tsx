import Axios from 'axios';
import React, {useEffect, useState} from 'react';
import {Button, Text, View} from 'react-native';
import {useDispatch, useSelector} from 'react-redux';
import {sayBye} from './utils';

function setMsgAction(msg) {
  return dispatch => {
    dispatch({type: 'setMsg', payload: msg});
  };
}

function callApi() {
  return dispatch => {
    Axios.request({url: 'some.url'}).then(res => {
      dispatch({type: 'setMsg', payload: res.data.msg});
    });
  };
}

const HomeScreen = () => {
  const [status, setStatus] = useState('');

  const state = useSelector(state => state.state);
  const dispatch = useDispatch();
  useEffect(() => {
    setTimeout(() => {
      dispatch(setMsgAction('action'));
    }, 10000);
  }, [dispatch]);
  return (
    <View style={{flex: 1, alignItems: 'center', justifyContent: 'center'}}>
      <Text testID="tv-test">{status}</Text>
      <Button
        testID="btn-test"
        onPress={() => {
          setStatus(sayBye());
        }}
        title="Update"
      />
    </View>
  );
};

//youtube: https://www.youtube.com/watch?v=6kM1Z5dDF-Q&ab_channel=BungFerdly
export default HomeScreen;
