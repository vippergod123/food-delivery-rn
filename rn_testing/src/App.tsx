import React, {useEffect, useState} from 'react';
import {Button, SafeAreaView, Text, View} from 'react-native';
import {Provider, useDispatch, useSelector} from 'react-redux';
import {applyMiddleware, createStore} from 'redux';
import HomeScreen from './HomeScreen';
import reducer from './reducer';
import {sayBye} from './utils';
import thunk from 'redux-thunk';
const store = createStore(reducer, applyMiddleware(thunk));

const App = () => {
  return (
    <Provider store={store}>
      <SafeAreaView style={{flex: 1, backgroundColor: 'white'}}>
        <HomeScreen />
      </SafeAreaView>
    </Provider>
  );
};

export default App;

//youtube: https://www.youtube.com/watch?v=6kM1Z5dDF-Q&ab_channel=BungFerdly
