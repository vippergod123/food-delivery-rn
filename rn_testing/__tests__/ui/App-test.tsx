/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../../src/App';

import renderer, {act, create} from 'react-test-renderer';
import {createStore} from 'redux';
import {Provider} from 'react-redux';
import reducer from '../../src/reducer';
import {sayBye} from '../../src/utils';
import Axios from 'axios';

const store = createStore(reducer, {msg: 'default msg!'});
const tree = create(
  <Provider store={store}>
    <App />
  </Provider>,
);

jest.mock('../../src/utils');

//test for change of ui
// 1. true -> ui not change at all
// 2. false -> ui change something -> fail test -> if changes acceptable -> npm test -u
test('snapshot', () => {
  expect(tree).toMatchSnapshot();
});

it('renders correctly', () => {
  renderer.create(<App />);
});

test('button pressed', () => {
  //mocking
  const mock = 'ABC';
  sayBye.mockReturnValue(mock);

  //verify
  const button = tree.root.findByProps({testID: 'btn-test'}).props;
  act(() => button.onPress());

  const text = tree.root.findByProps({testID: 'tv-test'}).props;
  expect(text.children).toEqual(mock);
});

test('async task', () => {
  //mocking
  const mock = 'ABC';
  sayBye.mockReturnValue(mock);

  //run
  act(() => jest.runAllTimers());

  //verify
  const text = tree.root.findByProps({testID: 'tv-test'}).props;
  expect(text.children).toEqual(mock);
});

test('reducer work properly', () => {
  expect(store.getState().msg).toEqual('default msg!');
});

test('test axios', () => {
  const mock = 'success';
  Axios.request.mockImplementation(() => Promise.resolve({data: {msg: mock}}));

  //run
  act(() => jest.runAllTimers());

  const text = tree.root.findByProps({testID: 'tv-test'}).props;
  expect(text.children).toEqual(mock);
});
