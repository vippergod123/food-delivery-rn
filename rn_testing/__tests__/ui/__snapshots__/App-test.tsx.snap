// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      testID="tv-test"
    >
      
    </Text>
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btn-test"
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Update
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
